sky/data/storage.py

```diff
@@ -1865,10 +1902,615 @@ def _delete_gcs_bucket(self, bucket_name: str) -> bool:
+
+    def mount_command(self, mount_path: str) -> str:
+        """Returns the command to mount the container to the mount_path.
+
+        Uses blobfuse2 to mount the container.
+
+        Args:
+            mount_path: str; Path to mount the container to
+
+        Returns:
+            str: a heredoc used to setup the AZ Container mount
+        """
+        install_cmd = mounting_utils.get_az_mount_install_cmd()
+        mount_cmd = mounting_utils.get_az_mount_cmd(self.bucket_name,
+                                                    self.storage_account_name,
+                                                    mount_path,
+                                                    self.storage_account_key)
+        return mounting_utils.get_mounting_command(mount_path, install_cmd,
+                                                   mount_cmd)
+
+    def _create_az_bucket(self, container_name: str) -> StorageHandle:
+        """Creates AZ Container.
+
+        Args:
+            bucket_name: str; Name of bucket
+            region: str; Region name, e.g. eastus, westus
+
+        Returns:
+            StorageHandle: handle to interact with the container
+
+        Raises:
+            StorageBucketCreateError: If container creation fails.
+        """
+        try:
+            container = self.storage_client.blob_containers.create(
+                self.resource_group_name, self.storage_account_name,
+                container_name, {})
+            logger.info('Created AZ Container '
+                        f'{container_name} in {self.region}.')
+        except azure.exceptions().ResourceExistsError as e:
+            error_msg, error_code = e.error.message, e.error.code
+            if error_code == 'ContainerOperationFailure':
+                if 'container is being deleted' in error_msg:
+                    with ux_utils.print_exception_no_traceback():
+                        raise exceptions.StorageBucketCreateError(
+                            f'The bucket {self.name!r} is currently being '
+                            'deleted. Please wait for the deletion to complete'
+                            'before attempting to create a bucket with the '
+                            'same name. This may take a few minutes. ')
+        return container
+
+    def _delete_az_bucket(self, container_name: str) -> bool:
+        """Deletes AZ Container, including all objects in Container.
+
+        Args:
+            container_name: str; Name of container
+
+        Returns:
+            bool: True if container was deleted, False if it's
+                deleted externally.
+        """
+        try:
+            with rich_utils.safe_status(
+                    f'[bold cyan]Deleting Azure cotainer {container_name}[/]'):
+                self.storage_client.blob_containers.get(
+                    self.resource_group_name,
+                    self.storage_account_name,
+                    container_name,
+                )
```

Comment:
why do we need to get it first?

Reply:
It's checking for the existance of the container before deletion. It causes an error if we attempt to delete when it does not exist. Left a comment on this.

Reply:
do we really need it? doesnt the following `self.storage_client.blob_containers.delete(` also raises similar exceptions?